### ISSUE TITLE
[YSQL] Query error: Restart read required

### ISSUE THREAD

**Issue by ryli17** [yugabyte/yugabyte-db#4205]
Yugabyte version: 2.1.3.0-b26

When I run 5 threads to do select/insert/update concurrently on a table, 200 records per thread, I got the following error:
`Query error: Restart read required at: { read: { physical: 1586581117344246 logical: 12 } local_limit: { physical: 1586581117344563 logical: 6 } global_limit: <min> in_txn_limit: <max> serial_no: 0 }`

It might be a regression. I didn't see it in v 2.1.2.

**Comment by frozenspider** [comment]
Hey @ryli17, could you please provide table DDL and queries you're running? If you have a sample code somewhere, it would be great!

**Comment by ryli17** [comment]
[multi_insert_update_client_pure.py.txt](https://github.com/yugabyte/yugabyte-db/files/4472637/multi_insert_update_client_pure.py.txt)

**Comment by ryli17** [comment]
Hi @frozenspider , The DDL please refer to https://github.com/yugabyte/yugabyte-db/issues/4215. The test script is uploaded as above. To repro it,
python3 multi_insert_pudate_client_pure.py --num_threads 5 --total_records 1000 --host <host>

**Comment by mbautin** [comment]
@ryli17: I'm using this schema to run the test: `create table ecx_pricing_service_provider (customer TEXT, ibx TEXT, cloudpackagetype TEXT, rate TEXT, virtualpresence TEXT, data1 TEXT, data2 TEXT, PRIMARY KEY (customer, ibx, cloudpackagetype, rate, virtualpresence));`. Please let me know if the schema you're using is different.

**Comment by ryli17** [comment]
@mbautin The schema I use is not the same. Please refer to https://github.com/yugabyte/yugabyte-db/issues/4215 for the schema. Thanks!

**Comment by mbautin** [comment]
@ryli17: thanks for sharing the schema. I've been running this workload in a loop on a 3-node cluster of c5.xlarge nodes on AWS, version 2.1.3.0 build 26 revision f0082094c4583989665b72e78180d5874600090e of YugabyteDB.

Example output of one iteration of the script:
```
** Started executing the benchmark test **
** Each record will be queried in db first, update the data if exists otherwise insert the record **
the time it took to process the 1000  records into the database is 1.0984179973602295
There were total 1000 select, 999 insert and 1 update operations
```

Total number of iterations so far:
```
$ grep "Started executing" workload.log | wc -l
275303
```

The error did occur. Total number of errors:
```
$ grep "Query error" workload.log | wc -l
15
```
(occurrence rate of ~0.0055%).

All iterations where the error happened: https://gist.githubusercontent.com/mbautin/8c0a96081cd3ac7f0f01a8f292f8d766/raw
(as produced by `grep -B 2 -A 4 "Query error" workload.log`).

We are looking more deeply into the root cause of this error. Similar "read restart" errors for SELECT statements should be retried automatically by our YSQL server. Because this error happens so rarely on this workload, I don't think it is a regression.

**Comment by mbautin** [comment]
@ryli17 In your observations, what was the frequency of this error?

**Comment by frozenspider** [comment]
This was caused by our inability to restart `EXECUTE select_stmt` issued in "simple" (`ysqlsh`-like) driver mode. Fixed in 581e952.
